skycatalogs/utils/config_utils.py

```diff
@@ -386,42 +446,221 @@ def assemble_file_metadata(pkg_root, inputs=None, run_options=None,
     ----------
-    object_types: iterable of (pointsource) object type names
+    inpath     string            path to file
+    silent     boolean           if file not found and silent is true,
+                                 return None.  Else raise exception
 
     Returns
     -------
-    A dict with object type names for keys.  Values are also dicts with
-    keys = model name and values defining struct of parameters for that
-    model, e.g. ordered dict with parameter names for keys and parameter
-    data types for values
+    dict representing contents of file, or None if silent and file not found
     '''
-    models = dict()
-    if 'agn' in object_types:
-        models['agn'] = _AGN_MODELS
-
-    return models
+    if resolve_include:
+        ldr = YamlIncludeLoader
+    else:
+        ldr = YamlPassthruIncludeLoader
+    try:
+        with open(inpath, mode='r') as f:
+            data = yaml.load(f, Loader=ldr)
+            return data
+    except FileNotFoundError as ex:
+        if silent:
+            return None
+        else:
+            raise ex
 
 
-def get_file_metadata(fpath, key='provenance'):
+class ConfigWriter():
     '''
-    Extract metadata from a parquet catalog file. Entries in the file
-    metadata are key-value pairs.  Return a dict with the specified
-    key as its only key
+    Saves a little context needed by utilities
     '''
-    meta = pq.read_table(fpath, columns=[]).schema.metadata
-    to_return = dict()
+    def __init__(self, skycatalog_root, catalog_dir, top_name,
+                 overwrite=False, logname=None):
+        self._skycatalog_root = skycatalog_root
+        self._catalog_dir = catalog_dir
+        self._out_dir = os.path.join(skycatalog_root, catalog_dir)
+        self._top_name = top_name
+        self._overwrite = overwrite
+        if not logname:
+            logname = 'skyCatalogs:ConfigUtils'
+        self._logger = logging.getLogger(logname)
+
+        # associate fragment name with config builder
+        self._make_frag = {'star': create_star_config,
+                           'galaxy': create_galaxy_config,
+                           'diffsky_galaxy': create_diffsky_galaxy_config,
+                           'sso': create_sso_config,
+                           'snana': create_snana_config,
+                           'gaia_star_butler': create_gaia_star_butler_config,
+                           'gaia_star_direct': create_gaia_star_direct_config}
+
+    def write_yaml(self, input_dict, outpath):
+        '''
+        Write yaml file if
+          * it doesn't already exist     or
+          * we're allowed to overwrite
+        '''
+        if not self._overwrite:
+            try:
+                with open(outpath, mode='x') as f:
+                    yaml.dump(input_dict, f)
+            except FileExistsError:
+                txt = 'write_yaml: Will not overwrite pre-existing config'
+                self._logger.warning(txt + outpath)
+                return
```

Comment:
It isn't used currently, but returning the path written seemed like it _might_ be useful.   In case of the exception nothing is written so `None` is returned.

Reply:
Depending on how a return value would be used, I would nominally expect that
```
try:
    with open(outpath, mode='x') as f:
        yaml.dump(input_dict, f)
except FileExistsError:
    txt = 'write_yaml: Will not overwrite pre-existing config'
    self._logger.warning(txt + outpath)

return outpath
```
has the desired behavior, since in both cases (i.e., with and without the exception), the path to the yaml file with the specified config is returned.

Reply:
Returning None is an indication (available programmatically, not just in a log message) that nothing got written

Reply:
Sure, but since nothing is using the return values, it's not clear that is or is not the right thing.